Refactor checkContainer to getContainer

DunderRoffe opened the pull request:
Refactor checkContainer(id, &container) to getContainer(id) which makes for cleaner code

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

Commit:
Refactor checkContainer to getContainer

Refactor checkContainer(id, &container) to getContainer(id) which makes for cleaner code

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

tobsan reviewed (approved):
It would be nicer with a shared_ptr instead of a raw pointer. Is that doable without changing too much of the code that uses getContainer?

thenor reviewed (approved):
Is this the entire 221 user story or only part of it?

Commit:
Use shared_ptr instead of unique_ptr in Agent

Use std::shared_ptr instead of unique_ptr to store and handle containers in
SoftwareContainerAgent as it makes more sense from a concurrency perspective.

Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>
